numpy/core/src/multiarray/typeinfo.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Provides namedtuples for numpy.core.multiarray.typeinfo
+ * Unfortunately, we need two different types to cover the cases where min/max
+ * do and do not appear in the tuple.
+ */
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+/* In python 2, this is not exported from Python.h */
+#include <structseq.h>
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+#include "npy_pycompat.h"
+
+
+PyTypeObject PyArray_typeinfoType;
+PyTypeObject PyArray_typeinforangedType;
+
+static PyStructSequence_Field typeinfo_fields[] = {
+    {"char",      "The character used to represent the type"},
+    {"num",       "The numeric id assigned to the type"},
+    {"bits",      "The number of bits in the type"},
+    {"alignment", "The alignment of the type in bytes"},
+    {"type",      "The python type object this info is about"},
+    {NULL, NULL,}
+};
+
+static PyStructSequence_Field typeinforanged_fields[] = {
+    {"char",      "The character used to represent the type"},
+    {"num",       "The numeric id assigned to the type"},
+    {"bits",      "The number of bits in the type"},
+    {"alignment", "The alignment of the type in bytes"},
+    {"max",       "The maximum value of this type"},
+    {"min",       "The minimum value of this type"},
+    {"type",      "The python type object this info is about"},
+    {NULL, NULL,}
+};
+
+static PyStructSequence_Desc typeinfo_desc = {
+    /* name          */ "numpy.core.multiarray.typeinfo",
```

Comment:
The comment would normally go after the entry.

Reply:
I think it would also look better, especially as it would be expected.

Reply:
Column alignment is easier to maintain this way, but I think you're right that consistency wins here, and cpython uses what you suggest internally